### PR TITLE
client: WithHTTPHeaders: improve doc, add test coverage, and error on duplicate headers

### PR DESCRIPTION
**Pull request by thaJeztah**
The behavior of this option was under-documented, and there's an issue with duplicate headers leading to unexpected behavior; changing that is probably not suitable for a patch release, so let's start with adding test-coverage, and adding a TODO for a future release.

**- What I did**

**- How I did it**

**- How to verify it**

**- Human readable description for the release notes**

```markdown changelog
Go SDK: client.WithHTTPHeaders now detects if duplicate headers are set, and produces an error. Previously, duplicate errors would be randomized, resulting in undefined behavior.
```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: client/client_options.go [hunk 1]
```diff
@@ -194,11 +194,25 @@ func WithUserAgent(ua string) Opt {
...
+		c.customHTTPHeaders = make(map[string]string)
+		for k, v := range headers {
+			k = http.CanonicalHeaderKey(k)
+			_, ok := c.customHTTPHeaders[k]
+			if ok {
+				// TODO(thaJeztah): consider returning an error on duplicates
+				continue
+			}
+			c.customHTTPHeaders[k] = v
+		}
```
Comment (thaJeztah) [inline client/client_options.go:214]: This is preserving the current behavior, but removing duplicates early; before this patch, duplicates would be silently discarded when we set the headers (`headers.Set` normalises the header-name, so overwriting earlier values); https://github.com/moby/moby/blob/086a6ebf79a7fadd2c3d010bec7b937b0b068739/client/request.go#L310-L314

File: client/client_options.go [hunk 2]
```diff
@@ -194,11 +194,25 @@ func WithUserAgent(ua string) Opt {
...
+// between runs due to map iteration order.
 func WithHTTPHeaders(headers map[string]string) Opt {
 	return func(c *clientConfig) error {
-		c.customHTTPHeaders = headers
+		c.customHTTPHeaders = make(map[string]string)
+		for k, v := range headers {
+			k = http.CanonicalHeaderKey(k)
+			_, ok := c.customHTTPHeaders[k]
+			if ok {
+				// TODO(thaJeztah): consider returning an error on duplicates
```
Comment (vvoland) [inline client/client_options.go]: So this doesn't change the current behavior and they are still discarded silently right?

So
```
Accept: application/json
Accept: text/plain
```

would silently still produce unexpected results.

I think we should just bite the bullet and return an error in this PR already.

Comment (thaJeztah) [reply client/client_options.go]: Yeah, didn't want to do it if it's a patch release; can do a quick follow-up as well to make it more visible

Comment (thaJeztah) [reply client/client_options.go]: do your example; it would be random when using mixed casing; `ACCEPT` and `Accept` (eg) in the same map provided here. Probably very corner case if someone would set that

Comment (thaJeztah) [reply client/client_options.go]: Updated; added an error if duplicates are detected.

**Review by vvoland — APPROVED** [review]
